src/DataStructures/VectorImpl.hpp

```diff
@@ -83,6 +83,10 @@ constexpr bool is_assignable_v = is_assignable<LhsDataType, RhsDataType>::value;
 /// `VectorImpl`
 struct MarkAsVectorImpl {};
 
+/// \ingroup DataStructuresGroup
+/// Default static size for vector impl
+constexpr size_t default_vector_impl_static_size = 5;
```

Comment:
Should we keep this at `0` for now? I'm not sure it will benefit us except maybe in 1d, and if we know we need it, we can use the `HybridVector` you said you'll add. It'll also help keep down memory usage. @wthrowe @kidder thoughts on this?

Reply:
I like this idea.